src/compiler/compile/render_ssr/handlers/Element.ts

```diff
@@ -164,6 +164,11 @@ export default function(node: Element, renderer: Renderer, options: RenderOption
 
 			renderer.add_expression(x`($$value => $$value === void 0 ? ${result} : $$value)(${node_contents})`);
 		} else {
+			if (node.name === 'textarea') {
+				// Restore the leading newline immediately after `<textarea>`.
+				// see https://html.spec.whatwg.org/multipage/syntax.html#element-restrictions
+				renderer.add_string('\n');
```

Comment:
this seemed to indiscriminately add newline even when the original code did not have a newline character

Reply:
Yes. That's right. The result displayed in the browser does not change, so there should be no problem other than the file size.
I thought that complicated source code was needed to reduce unnecessary line feed codes. Do you know a good way?

Reply:
what i meant is that in line 180, for `<pre>`, you only prepended `'\n'` if the `first.data` starts with newline, i believe `<textarea>` need a similar condition right?

in the `server-side-rendering/samples/textarea-content` test case, the `#textarea-without-leading-newline` textarea didnt have newline in the source but ended up having newlines in the rendered HTML instead

Reply:
Oh thank you. I understand what you say. I thought I had to use node_contents to determine the starts with newline, but as you say it just accesses the original attribute node.